Lib/fontTools/cffLib/specializer.py

```diff
@@ -26,13 +27,15 @@ def programToString(program):
 	return ' '.join(str(x) for x in program)
 
 
-def programToCommands(program):
+def programToCommands(program, numRegions=None, **kwargs):
```

Comment:
The extra args are there because of specializeProgram(program, **kwargs) and generalizeProgram(program, **kwargs). **kwargs needs to hold both numRegions  and generalizeFirst. If these args  are both in **kwargs, then programToCommnd and specializeCommand complain about the unused arg unless you provide the additional **kwargs to swallow the unused arg. I don't really like what I did, but it is better than the alternatives I thought of.

Reply:
Why not just spell out the arguments that are needed?

Reply:
Because I saw that the prior code used **kwargs rather than the arguments actually needed. I assumed this was a deliberate choice. I do see that it allows for adding future new arguments to the called functions, without having to change specializeProgram() and generalizeProgram(). That said, I am perfectly happy to follow your suggestion. I would then change **kwargs in specializeProgram() and generalizeProgram  to the two args, and pass only the needed args to the callees. Just confirm that this is what you would prefer.

Reply:
I see.  That definitely explains your choice.  Thank you.

Generally I like using kwargs for methods that just pass along arguments, but not in methods that consume them.  What I like to see here, I think, is to keep things as they are, just add `numRegion` anywhere it's needed.  You can add `numRegions=None` just before `**kwargs` in the methods that take that.  This will separate that one argument from the rest, then you decide which methods you pass the numRegions to and which one the `**kwargs`.  I *think* that works but up to you whichever works.  Thanks.